### ISSUE TITLE
API - Zone PUT does not empty master field

### ISSUE THREAD

**Issue by geoff-maddock** [PowerDNS/pdns#9939]
- Program: Authoritative
 - Issue type: Bug report

### Short description
When making a PUT request to update a ZONE via the API, the master field does not empty

### Environment
 - Operating system: unix
 - Software version:   4.1.14
 - Software source: unknown

### Steps to reproduce
Create a zone that has a master:
```
{
  "name": "slavezone.com",
  "kind": "Slave",
  "type": "Zone",
  "masters":  ["10.0.0.1"]
}
```

Make a future PUT request to the same zone ID with an empty master
```
{
  "name": "slavezone.com",
  "kind": "Slave",
  "type": "Zone",
  "masters": []
}
```

Get the same zone and note that the master IP has not changed from 10.0.0.1

### Expected behaviour
The master IP should be emptied on the PUT request.

### Actual behaviour
The master IP is not removed

### Other information
I couldn't find any other information about whether this might be expected behavior.

**Comment by Habbie** [comment]
Hello, thank you for your report! You are running version 4.1, which is so old we do not even provide security patches for it any more. Can you please try 4.4?

**Comment by Habbie** [comment]
>   "masters": "[10.0.0.1]"

Also, this syntax looks wrong to me. Should be `["10.0.0.1"]` I believe.

**Comment by geoff-maddock** [comment]
> > "masters": "[10.0.0.1]"
> 
> Also, this syntax looks wrong to me. Should be `["10.0.0.1"]` I believe.

Looked back at the code - this is how we're doing it, it was my mistake in typing out the issue.

**Comment by Habbie** [comment]
> Looked back at the code - this is how we're doing it, it was my mistake in typing out the issue.

Please use copy/paste next time to avoid wasting time like this.

**Comment by geoff-maddock** [comment]
Sorry about that.  Updated the issue to reflect the experienced bug.

**Comment by Habbie** [comment]
I've tested this on current master (which is close to 4.4.0) and the `masters` field gets emptied correctly, so I'm closing this in the assumption that we fixed it somewhere between 4.1.14 and 4.4.0.

**Comment by geoff-maddock** [comment]
Thanks for the update.
